### PR TITLE
Make Bandwidth Agreements Secure / Trustable

### PR DESCRIPTION
**Pull request by wthorp**
Goals of this PR:
Fix Bandwidth agreements so that a storage node can trust them.
Right now only the satellite can trust them.

### CODE REVIEW

File: pkg/bwagreement/server.go [hunk 1]
```diff
@@ -12,10 +12,45 @@ import (
...
+	//UpPrivKey wraps errors related to uplink private keys
+	UpPrivKey = errs.Class("Processing uplink private key")
+	//Renter wraps errors related to renter bandwidth allocations
+	Renter = errs.Class("Renter agreement")
+	//Payer wraps errors related to payer bandwidth allocations
+	Payer = errs.Class("Payer agreement")
+	//Missing indicates missing or empty information
+	Missing = errs.Class("Required field is empty")
+	//Ecdsa indicates a key was not an ECDSA key
+	Ecdsa = errs.New("Key is not ecdsa key")
```
Comment (kaloyan-raev) [inline pkg/bwagreement/server.go]: This should be all caps as it is an acronym.

File: pkg/bwagreement/test/server_test.go [hunk 2]
```diff
@@ -6,19 +6,26 @@ package test
...
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
 
 	"storj.io/storj/internal/testcontext"
 	"storj.io/storj/internal/testidentity"
-	"storj.io/storj/internal/teststorj"
 	"storj.io/storj/pkg/bwagreement"
+	"storj.io/storj/pkg/identity"
 	"storj.io/storj/pkg/pb"
+	e "storj.io/storj/pkg/pb"
```
Comment (egonelbre) [inline pkg/bwagreement/test/server_test.go]: Duplicate import

File: pkg/piecestore/psserver/server_test.go [hunk 3]
```diff
@@ -260,7 +252,7 @@ func TestRetrieve(t *testing.T) {
 					}),
 				}
 
-				s, err := cryptopasta.Sign(ba.Data, TS.k.(*ecdsa.PrivateKey))
+				s, err := cryptopasta.Sign(ba.Data, upID.Key.(*ecdsa.PrivateKey))
 				assert.NoError(err)
 				ba.Signature = s
```
Comment (littleskunk) [inline pkg/piecestore/psserver/server_test.go]: `ba.PubKey` is missing. How can this test case work if it send invalid bwagreements? Maybe the storage node is not validating them?

Comment (wthorp) [reply pkg/piecestore/psserver/server_test.go]: given the information you've presented, I'd say the storage node is not validating them.  now I need to figure out why.

Comment (littleskunk) [reply pkg/piecestore/psserver/server_test.go]: The complete testset was disabled. I enabled them and was able to fix them.

File: pkg/piecestore/psserver/server_test.go [hunk 4]
```diff
@@ -649,95 +591,45 @@ func newTestServerStruct(t *testing.T, ids []storj.NodeID) (*Server, func()) {
...
-			t.Fatal(serr)
-		}
-		// TODO:fix this error check
-		_ = os.RemoveAll(tmp)
-		// if err := os.RemoveAll(tmp); err != nil {
-		// 	t.Fatal(err)
-		// }
+	//init ps server grpc
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	assert.NoError(t, err)
```
Comment (jtolds) [inline pkg/piecestore/psserver/server_test.go]: not a big deal but you probably want require.NoError for this type of code style. assert.NoError marks the test as failed but lets the test keep going. require.NoError halts the test.

Comment (jtolds) [reply pkg/piecestore/psserver/server_test.go]: i do see that this affects most of this test suite

Comment (wthorp) [reply pkg/piecestore/psserver/server_test.go]: Ah!  I didn't know about that one

File: pkg/pb/BandwidthUtils.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	} else if sm.GetSignature() == nil {
+		return false, Missing.New("message signature")
+	} else if sm.GetCerts() == nil {
+		return false, Missing.New("message certificates")
+	}
+	return true, nil
+}
+
+//Unpack helps get things out of a RenterBandwidthAllocation
+func (rba *RenterBandwidthAllocation) Unpack() (*RenterBandwidthAllocation_Data, *PayerBandwidthAllocation, *PayerBandwidthAllocation_Data, error) {
```
Comment (jtolds) [inline pkg/pb/BandwidthUtils.go]: should unpack check the signatures just so it's less easy to forget to do so in the future?

Comment (wthorp) [reply pkg/pb/BandwidthUtils.go]: In cases like audit, the signatures have already been validated.

**Pull request merged**